addons/udes_security/models/blocked_file_type.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class BlockedFileType(models.Model):
+    """
+    Blocked file type which the user is prevented from uploading or downloading
+    """
+
+    _name = "udes.blocked_file_type"
+    _description = "Blocked File Type"
+    _order = "name"
+
+    _sql_constraints = [("name_uniq", "unique (name)", "The File Type must be unique")]
+
+    name = fields.Char("File Type", required=True, index=True)
+    description = fields.Char("Description")
+    active = fields.Boolean("Active?", default=True)
+
+    def _format_name(self, name):
+        """Convert supplied name to lowercase if needed"""
+        formatted_name = name.lower().replace(".", "") if name else ""
+        return formatted_name
+
+    def _format_vals(self, vals):
+        """Format the required values in the supplied dictionary and 
+           return the updated values"""
+        formatted_vals = {}
+
+        name = vals.get("name", "")
+        if name:
+            name = self._format_name(name)
+            formatted_vals["name"] = name
+
+        return formatted_vals
+
+    def _update_attachments_active_status(self, deleted=False):
+        """Set active field on attachments that match the supplied blocked file types"""
+        IrAttachment = self.env["ir.attachment"]
+
+        attachments_set_active = IrAttachment.browse()
+        attachments_set_inactive = IrAttachment.browse()
+
+        for blocked_file_type in self:
+            blocked_file_type_active = blocked_file_type.active
```

Comment:
nit: What about renaming it to is_blocked_file_type ?

Reply:
Have now renamed this.